apps/coordinator/ts/scheduler/__tests__/scheduler.controller.test.ts

```diff
@@ -0,0 +1,111 @@
+import { EMode } from "@maci-protocol/sdk";
+import { HttpStatus } from "@nestjs/common";
+
+import { ESupportedNetworks } from "../../common";
+import { IdentityScheduledPollDto, SchedulePollWithSignerDto } from "../dto";
+import { SchedulerController } from "../scheduler.controller";
+import { SchedulerService } from "../scheduler.service";
+
+const scheduledPoll: SchedulePollWithSignerDto = {
+  maciAddress: "0x0",
+  pollId: 5,
+  mode: EMode.NON_QV,
+  chain: ESupportedNetworks.OPTIMISM_SEPOLIA,
+  deploymentBlockNumber: 1,
+};
+
+const identityScheduledPoll = {
+  maciAddress: scheduledPoll.maciAddress,
+  pollId: scheduledPoll.pollId,
+  chain: scheduledPoll.chain,
+} as IdentityScheduledPollDto;
+
+describe("SchedulerController", () => {
+  let service: jest.Mocked<SchedulerService>;
+
+  let controller: SchedulerController;
+
+  beforeEach(() => {
+    service = {
+      registerPoll: jest.fn().mockResolvedValue({ isScheduled: true }),
+      isPollScheduled: jest.fn().mockResolvedValue({ isScheduled: true }),
+      deleteScheduledPoll: jest.fn().mockResolvedValue({ isScheduled: false }),
+    } as unknown as jest.Mocked<SchedulerService>;
+
+    controller = new SchedulerController(service);
+  });
+
+  describe("register", () => {
+    it("should register a poll for finalization", async () => {
+      const { isScheduled } = await controller.register(scheduledPoll);
+
+      expect(isScheduled).toBe(true);
+    });
+
+    it("should throw an error if registration fails", async () => {
+      const errorMessage = "Registration failed";
+      service.registerPoll.mockRejectedValueOnce(new Error(errorMessage));
+
+      try {
+        await controller.register(scheduledPoll);
+
+        fail("Expected error was not thrown");
+      } catch (error: unknown) {
+        const returnedError = error as Error;
+
+        expect(returnedError.message).toBe(errorMessage);
+        expect(returnedError).toHaveProperty("status", HttpStatus.BAD_REQUEST);
+      }
```

Comment:
Why do you use try/catch here? It's possible to check rejected values with jest.

Reply:
it was because when using `await expect(controller.register(scheduledPoll).rejects.Throw(...)` I could not chain together `.reject.Throw(errorMessage).toHaveProperty("status", HttpStatus.BAD_REQUEST)`

I decided to go this way: one call, one fail and 2 expects.

I could do: 2 calls, 2 fails and 2 expects if it helps readibility 😊

Reply:
I was able to restructure it different and more clean. It is using the same as https://github.com/privacy-scaling-explorations/maci/blob/3b89cb83f2a10f890622f60097055115dda96777/apps/coordinator/ts/proof/__tests__/proof.controller.test.ts#L114